app/src/main/java/com/pusher/demo/features/marketplace/chat/MarketplaceChatPresenter.kt

```diff
@@ -109,13 +109,32 @@ class MarketplaceChatPresenter :  BasePresenter<MarketplaceChatPresenter.View>()
     }
 
     private fun getOtherMemberReadCursor(room: Room, otherMember : User) {
-        val readCursor = currentUser().getReadCursor(room, otherMember).successOrThrow()
-        view?.onOtherMemberReadCursorChanged(readCursor.position)
+        currentUser().getReadCursor(room, otherMember).fold(
+            onFailure = {
+                // this situation can happen when the room has just been created, and the other
+                // persons cursor has not yet been created as they have not yet read any messages.
+                Log.d(ChatkitManager.LOG_TAG, "failed to create the other members cursor")
+            },
+            onSuccess = {
```

Comment:
By a convention happy-path should be placed first (like in try-catch). I know that's what auto-complete suggests as it's `onFailure` first in `fold` signature (something we could consider swapping for v2), however as we use named params at the call site I think it would be better for `onSuccess` to go first.

Reply:
Makes sense! Should I add that to my comment PR in chatkit-android so we don't forget it there too?

Reply:
yes please, good call!